### DISCUSSION TITLE
I can't save a model as a sbml file

### DISCUSSION THREAD

**Discussion by gcxxx** [opencobra/cobrapy#1219] [Q&A]
Hi,
I create my own model 'on the fly' using cobrapy package. Now I want to save it as a sbml file, but an error occured, how to solve this problem? (By the way, I can save this model as json file, no error occured.) 
I upload my model in json file.
![image](https://user-images.githubusercontent.com/86390409/167291049-2d9f6d13-0f76-4d4e-8caa-7995a8d32ef3.png)
[133reactions.zip](https://github.com/opencobra/cobrapy/files/8646770/133reactions.zip)

**Comment by Midnighter** [comment]
SBML has more consistency checks than JSON. My guess is that some metabolite's compartment is set to `None` where SBML expects a compartment identifier. That's an issue with our SBML writer. Until it's fixed, maybe you can set it to an empty string or better yet, add a compartment to your model.
